### PR TITLE
Fix #1239 UX improvement for showing a user's reputations

### LINKED ISSUE #1239: UX improvement for showing a user's reputations
Even if a post a user got a reputation for has been deleted the links to the posts are still shown, so "username's post" leads to the post and "in threadname" leads to the corresponding thread (thread link is hidden for hard deleted posts). 

Using "Rating given for deleted post in threadname" or using "Rating given for deleted post in deleted thread" would increase the consistency of the display style (always show post and thread info) and would improve the user experience, because there are no links to content that is not available any more (or to this user in case of a soft deleted post). 

This issue is especially annoying for an user without the right to see soft deleted content (see http://community.mybb.com/thread-156131.html)

Original thread: [UX improvement for showing a user's reputations](http://community.mybb.com/thread-156133.html)

### PR DESCRIPTION
**Pull request by Sama34**
Fix #1239 UX improvement for showing a user's reputations

### CODE REVIEW

File: reputation.php [hunk 1]
```diff
@@ -936,21 +970,16 @@
...
 
 		// Is this rating specific to a post?
-		$postrep_given = '';
+		$postrep_given = $lang->sprintf($lang->postrep_given_nolink, $user['username']);
 		if($reputation_vote['pid'])
 		{
-			$link = get_post_link($reputation_vote['pid'])."#pid{$reputation_vote['pid']}";
-
-			$thread_link = '';
-			if(isset($post_reputation[$reputation_vote['pid']]))
```
Comment (Destroy666x) [inline reputation.php]: Why was `isset()` check removed?

File: reputation.php [hunk 2]
```diff
@@ -864,17 +864,51 @@
...
+		}
+
+		// get inactive forums
+		$inactive = get_inactive_forums();
+		if($inactive)
+		{
+			$sql[] = "p.fid NOT IN ({$inactive})";
+		}
+
+		if(!$mybb->user['ismoderator'])
```
Comment (Destroy666x) [inline reputation.php]: I don't think this part is needed with the 2 conditionals below.

Also, the `canonlyviewownthreads` permission check is still missing,

File: reputation.php [hunk 3]
```diff
@@ -936,21 +970,16 @@
 		$last_updated = $lang->sprintf($lang->last_updated, $last_updated_date);
 
 		// Is this rating specific to a post?
-		$postrep_given = '';
+		$postrep_given = $lang->sprintf($lang->postrep_given_nolink, $user['username']);
```
Comment (Destroy666x) [inline reputation.php]: This + lack of `isset()` introduce even more noticeable bugs than the fixed issue - every non-post reputation has now incorrect `Rating given for xxx's post` note and the post reputations which should be hidden lead to blank posts.

File: reputation.php (hunk 2 above)
Comment (Sama34) [inline reputation.php]: Doesn't `get_unviewable_forums()` handles `canonlyviewownthreads`?

Comment (Destroy666x) [inline reputation.php]: It never did, only `canviewthreads`

Comment (Sama34) [inline reputation.php]: I pushed a change to should handle those.

Comment (Sama34) [inline reputation.php]: > I don't think it makes the code faster either.

For posts may be not, but threads don't have an index on `visible` so I just added it.

**Pull request merged**